src/waveform/waveformwidgetfactory.cpp

```diff
@@ -565,7 +566,7 @@ void WaveformWidgetFactory::swap() {
             for (int i = 0; i < m_waveformWidgetHolders.size(); i++) {
                 WaveformWidgetAbstract* pWaveformWidget = m_waveformWidgetHolders[i].m_waveformWidget;
                 if (pWaveformWidget->getWidth() > 0) {
-                    QGLWidget* glw = dynamic_cast<QGLWidget*>(pWaveformWidget->getWidget());
+                    QOpenGLWidget* glw = dynamic_cast<QOpenGLWidget*>(pWaveformWidget->getWidget());
                     // Don't swap invalid / invisible widgets or widgets with an
                     // unexposed window. Prevents continuous log spew of
                     // "QOpenGLContext::swapBuffers() called with non-exposed
```

Comment:
Replacing `VSyncThread::swapGl(glw, i;)` below with `glw->context()->swapBuffers(window);` does what was done before by `VSyncThread::swapGl`, but it does not solve the jerky waveform scrolling. (GitHub won't let me comment directly on that line because you have not changed it.)